docs: add docker compose config example (#6554)
apollographql/router: docs/source/routing/self-hosted/containerization/docker.mdx

To run the router, your Docker container must have the [`APOLLO_GRAPH_REF`](/router/configuration/overview#apollo_graph_ref) and [`APOLLO_KEY`](/router/configuration/overview#apollo_key) environment variables set to your graph ref and API key, respectively.

> Below is a basic example of running a router image in Docker. It downloads your supergraph schema from Apollo and uses a default configuration that listens for connections on port `4000`.
> 
> You can use `docker run` with the following example command:

Reviewer: The "Below is a basic example" + new paragraph "You can use `docker run`..." threw me off a little. I was like, where is the basic example, before realizing the `docker run` example was the basic example.

```suggestion
Below is a basic example of using `docker run` to run a router image. It downloads your supergraph schema from Apollo and uses a default configuration that listens for connections on port `4000`.
```

Author: Will clarify that `docker run` and `docker compose` run the sample example:

```suggestion
Below is a basic example of running a router image with Docker, either with `docker run` or `docker compose`. It downloads your supergraph schema from Apollo and uses a default configuration that listens for connections on port `4000`.

You can use `docker run` with the following example command:
```

  ghcr.io/apollographql/router:<router-image-version>
```

You can also use `docker compose` with the following example `compose.yaml`:

```yaml title="compose.yaml"
services:
  apollo-router:
    image: ghcr.io/apollographql/router:<router-image-version>
    ports:
      - "4000:4000"
    environment:
      APOLLO_GRAPH_REF: "<your-graph-ref>"
      APOLLO_KEY: "<your-graph-api-key>"
```

> Make sure to replace `<router-image-version>` with whichever version you want to use, such as `v1.58.0`, and `<your-graph-ref>` and `<your-graph-api-key>` with your graph reference and API key, respectively.

Reviewer: ```suggestion
Whether you use `docker run` or `docker compose`, make sure to replace `<router-image-version>` with whichever version you want to use, such as `v1.58.0`, and `<your-graph-ref>` and `<your-graph-api-key>` with your graph reference and API key, respectively.
```
